### PR TITLE
#50 - Initial Doxygen Creation

### PR DESCRIPTION
**Pull request by jrdike**

### CODE REVIEW

File: docs/doxygen-config [hunk 1]
```diff
@@ -0,0 +1,2769 @@
...
+# the logo to the output directory.
+
+PROJECT_LOGO           =
+
+# The OUTPUT_DIRECTORY tag is used to specify the (relative or absolute) path
+# into which the generated documentation will be written. If a relative path is
+# entered, it will be relative to the location where doxygen was started. If
+# left blank the current directory will be used.
+
+OUTPUT_DIRECTORY       = C:/Users/jrd64/Documents/GitHub/Team-2135-Graphics-Applications-for-Transparent-Electronic-Displays/docs
```
Comment (Russell-Newton) [inline docs/doxygen-config:71]: We need to use a relative path here.

**Comment by Russell-Newton** [comment]
It would be great if we could add the html folder to a `.gitignore` file within the `docs/` folder. In order to actually have these ignored, you'd need to make a commit deleting the html folder. Then you can add the `.gitignore` and rebuild the html with doxygen. If the html folder is correctly ignored, it won't be prompted by GitHub Dekstop to be added to the commit

**Comment by xorbit3024** [comment]
Search bar does not respond. I can type in it, but pressing the search button or pressing the enter button on my keyboard both do nothing.

**Comment by Russell-Newton** [comment]
> It would be great if we could add the html folder to a `.gitignore` file within the `docs/` folder. In order to actually have these ignored, you'd need to make a commit deleting the html folder. Then you can add the `.gitignore` and rebuild the html with doxygen. If the html folder is correctly ignored, it won't be prompted by GitHub Dekstop to be added to the commit

You'd also probably need to update the docs readme with instructions for how to generate the doxygen project. Once we have all this, we can look into getting a CICD workflow that generates the doxygen for us and hosts it on github pages

**Comment by Russell-Newton** [comment]
> Search bar does not respond. I can type in it, but pressing the search button or pressing the enter button on my keyboard both do nothing.

We probably want to get a fix in under a different ticket. We probably don't want this to be a blocker for this PR.

**Review by Russell-Newton — COMMENTED** [review]
This is a good starting point. From here, we can look at generating this with a workflow

File: doxygen-config [hunk 2]
```diff
@@ -0,0 +1,2769 @@
...
+# Configuration options related to the input files
+#---------------------------------------------------------------------------
+
+# The INPUT tag is used to specify the files and/or directories that contain
+# documented source files. You may enter file names like myfile.cpp or
+# directories like /usr/src/myproject. Separate the files or directories with
+# spaces. See also FILE_PATTERNS and EXTENSION_MAPPING
+# Note: If this tag is empty the current directory is searched.
+
+INPUT                  = ./gesturelibrary/src
```
Comment (Russell-Newton) [inline doxygen-config:911]: Make this `./gesturelibrary/src ./gesturelibrary/includes`

**Review by xorbit3024 — APPROVED** [review]

**Review by Russell-Newton — APPROVED** [review]
Now that we're including both src and includes within the docs, this is looking good. Once we have this, we can make a workflow to deploy the pages automatically.

**Review by deborahsrcho — APPROVED** [review]

**Pull request merged**